src/renderer/kube-object.store.ts

```diff
@@ -281,21 +281,38 @@ export abstract class KubeObjectStore<T extends KubeObject = any> extends ItemSt
 
   subscribe(apis = this.getSubscribeApis()) {
     const abortController = new AbortController();
+    const [clusterScopedApis, namespaceScopedApis] = bifurcateArray(apis, api => api.isNamespaced);
 
-    // This waits for the context and namespaces to be ready or fails fast if the disposer is called
-    Promise.race([rejectPromiseBy(abortController.signal), Promise.all([this.contextReady, this.namespacesReady])])
-      .then(() => {
-        if (this.context.cluster.isGlobalWatchEnabled && this.loadedNamespaces.length === 0) {
-          apis.forEach(api => this.watchNamespace(api, "", abortController));
-        } else {
-          apis.forEach(api => {
-            this.loadedNamespaces.forEach((namespace) => {
-              this.watchNamespace(api, namespace, abortController);
-            });
-          });
-        }
-      })
-      .catch(noop); // ignore DOMExceptions
+    console.log([clusterScopedApis, namespaceScopedApis]);
+
+    for (const api of namespaceScopedApis) {
+      const store = apiManager.getStore(api);
+
+      // This waits for the context and namespaces to be ready or fails fast if the disposer is called
+      Promise.race([rejectPromiseBy(abortController.signal), Promise.all([store.contextReady, store.namespacesReady])])
```

Comment:
It was also a bug that this race was performed only for the `this` store?

Reply:
Yes I think so. Though I now that I think about it. Subscribing really should be reactive on the change set of selected namespaces (adding and removing the watches as the namespaces get selected and deselected).

But I think that is out of scope of this fix.